### PR TITLE
[dist] Add old test suite to docker

### PR DESCRIPTION
**Pull request by eduardoj**
First attempt to include old test suite into docker.

- Create following rake tasks:
  - `docker:test:old:prepare`
  - `docker:test:old:api`
  - `docker:test:old:webui`
- Create a specific docker containter: `Dockerfile.old_test_suite`.
- Change the `dev:prepare` and `dev:boostrap` rake tasks adding a non required parameter to deal with the old test suite.
- Add connection error types that only happen in docker when waiting for services to be ready.
- Fix relative path addition.

### CODE REVIEW

File: Dockerfile.old_test_suite [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+# Install requirements for the backend
+RUN zypper -n install --no-recommends --replacefiles \
+  obs-server obs-signd \
+  obs-service-download_src_package obs-service-download_files \
+  obs-service-download_url \
+  obs-service-format_spec_file obs-service-kiwi_import \
+  perl-Devel-Cover \
+  osc
+
+# Add our sign configuration
```
Comment (hennevogel) [inline Dockerfile.old_test_suite]: We don't need the signer in the old test suite...

File: Dockerfile.old_test_suite [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+RUN zypper -n install --no-recommends --replacefiles memcached
+
+# Install requirements for the backend
+RUN zypper -n install --no-recommends --replacefiles \
+  obs-server obs-signd \
+  obs-service-download_src_package obs-service-download_files \
+  obs-service-download_url \
+  obs-service-format_spec_file obs-service-kiwi_import \
+  perl-Devel-Cover \
+  osc
```
Comment (hennevogel) [inline Dockerfile.old_test_suite]: I must say I'm not too fond of just copying every `RUN` into this Dockerfile. We are going to live with this setup for quite some time (until we switched everything to rspec) and this means there are always two places I have to adopt for software/configuration changes. 

How about moving this (back) into a script that you can call with arguments. Something like

```
RUN contrib/docker-bootstrap.sh backend
```

to do the backend part

```
RUN contrib/docker-bootstrap.sh signer
```

to do the signer part etc.

The docker images then can call this script in any sequence they need and we have only one place we need to adopt.

File: Rakefile [hunk 3]
```diff
@@ -41,6 +41,33 @@ namespace :docker do
         sh "docker-compose -f docker-compose.ci.yml stop"
       end
     end
+
+    namespace :old do
+      desc 'Wake up database and prepare docker image for running an old test suite'
+      task :prepare do
+        sh "docker-compose up -d db"
+        sh "docker build . -t openbuildservice/old_test_suite -f Dockerfile.old_test_suite --build-arg CONTAINER_USERID=#{CONTAINER_USERID}"
```
Comment (hennevogel) [inline Rakefile]: Why would I have to build this pretty large image every time?

**Comment by DavidKang** [comment]
@eduardoj, what is the status of this PR?

**Comment by eduardoj** [comment]
@DavidKang, a little bit stalled.

File: Dockerfile.old_test_suite (hunk 2 above)
Comment (bgeuken) [reply Dockerfile.old_test_suite]: Wouldn't the docker way be to have a docker file that does the common task and the use that image in any other docker files that's needs these packages?

File: Dockerfile.backend [hunk 4]
```diff
@@ -3,16 +3,10 @@ FROM openbuildservice/base
...
-  obs-server obs-signd \
-  obs-service-download_src_package obs-service-download_files \
-  obs-service-download_url \
-  obs-service-format_spec_file obs-service-kiwi_import \
-  perl-Devel-Cover \
-  osc
+RUN /root/bin/docker-bootstrap.sh backend
 
 # Add our sign configuration
+RUN zypper -n install --no-recommends --replacefiles obs-signd
```
Comment (hennevogel) [inline Dockerfile.backend]: 💭 Why is this not part of the bootstrap script?

File: contrib/docker-bootstrap.sh [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    memcached)
+      zypper -n install --no-recommends --replacefiles memcached
+      ;;
+
+  esac
+}
+
+for option in "$@"
+do
+  case "${option}" in
```
Comment (hennevogel) [inline contrib/docker-bootstrap.sh]: 💭 Why 2 `case` loops?

File: Dockerfile.423 [hunk 6]
```diff
@@ -1,5 +1,9 @@
 FROM opensuse:42.3
 
+# Temporary fix pam
```
[the remark this answers is not in the document]
Comment (eduardoj) [reply Dockerfile.423]: Changed.

File: Dockerfile.backend (hunk 4 above)
Comment (eduardoj) [reply Dockerfile.backend]: Moved to the bootstrap script.

File: contrib/docker-bootstrap.sh (hunk 5 above)
Comment (eduardoj) [reply contrib/docker-bootstrap.sh]: Changed. Used only one loop.

**Pull request merged**